event_session/models/event_mail.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2017 David Vidal<david.vidal@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from odoo.addons.event.models.event_mail import _INTERVALS
+except ImportError:
+    _logger.debug("Can not import events module.")
+
+
+class EventMailScheduler(models.Model):
+    _inherit = "event.mail"
+
+    event_id = fields.Many2one(required=False)
+    session_id = fields.Many2one(
+        comodel_name="event.session",
+        string="Session",
+        ondelete="cascade",
+    )
+    event_mail_template_id = fields.Many2one(
+        comodel_name="event.mail.template",
+        string="Event Mail Template",
+        ondelete="cascade",
+    )
+
+    @api.depends(
+        "mail_sent",
+        "interval_type",
+        "event_id.registration_ids",
+        "mail_registration_ids",
+    )
+    def _compute_done(self):
+        super()._compute_done()
+        for event_mail in self:
+            if event_mail.session_id and event_mail.interval_type not in [
+                "before_event",
+                "after_event",
+            ]:
+                event_mail.done = (
+                    True
+                    if event_mail.event_id.sessions_count > 0
+                    and not event_mail.session_id
+                    else len(event_mail.mail_registration_ids)
+                    == len(event_mail.session_id.registration_ids)
+                    and all(line.mail_sent for line in event_mail.mail_registration_ids)
+                )
+
+    @api.depends(
+        "event_id.state",
```

Comment:
```suggestion
        "event_id.stage_id",
```
That's why travis is red

Reply:
Weird, that's wasn't on my local code. I'm pushing again anyway